src/clojush/random.clj

```diff
@@ -38,50 +38,84 @@
         (recur (inc parens)
                (rest probabilities))))))
 
+(defn conditional-thread
```

Comment:
Did you invent this, or does this come from somewhere? 

Am I correct in thinking you could use `cond->` for this, but you'd end up with a bunch of repetition since you'd have to repeat each expression as both the "test" and the "result"?

Does it belong here, or in the `util`?